compiler.cmake

```diff
@@ -19,51 +19,140 @@ macro(_SETUP_PROJECT_WARNINGS)
-    set(WARNING_CXX_FLAGS "${WARNING_CXX_FLAGS} /wd4275")
-    # C4355 "this" used in base member initializer list
-    set(WARNING_CXX_FLAGS "${WARNING_CXX_FLAGS} /wd4355")
-  endif()
+  set(MSVC_FLAGS
+      -D_SCL_SECURE_NO_WARNINGS
+      -D_CRT_SECURE_NO_WARNINGS
+      -D_CRT_SECURE_NO_DEPRECATE
+      # -- The following warnings are removed to highlight the output C4101 The
+      # local variable is never used removed since happens frequently in
+      # headers.
+      /wd4101
+      # C4250 'class1' : inherits 'class2::member' via dominance
+      /wd4250
+      # C4251 class 'type' needs to have dll-interface to be used by clients of
+      # class 'type2' ~ in practice, raised by the classes that have non-dll
+      # attribute (such as std::vector)
+      /wd4251
+      # C4275 non - DLL-interface used as base for DLL-interface
+      /wd4275
+      # C4355 "this" used in base member initializer list
+      /wd4355)
+
+  cxx_flags_by_compiler_frontend(
+    GNU
+    ${GNU_FLAGS}
+    MSVC
+    ${MSVC_FLAGS}
+    OUTPUT
+    WARNING_CXX_FLAGS_LIST
+    FILTER)
+  string(REPLACE ";" " " WARNING_CXX_FLAGS "${WARNING_CXX_FLAGS_LIST}")
 
   set(CMAKE_CXX_FLAGS "${WARNING_CXX_FLAGS} ${CMAKE_CXX_FLAGS}")
 
   list(APPEND LOGGING_WATCHED_VARIABLES WARNING_CXX_FLAGS)
 endmacro(_SETUP_PROJECT_WARNINGS)
+
+#[=======================================================================[.rst:
+.. command:: CXX_FLAGS_BY_COMPILER_FRONTEND(<GCC   [<flags1>...]>
+                                            <MSVC  [<flags1>...]>
+                                            OUTPUT flags
+                                            <FILTER>)
+
+Detect the compiler frontend (the command line interface) and output the
+corresponding ``CXX_FLAGS``.
+
+The following arguments allow to specify ``CXX_FLAGS`` for a compiler
+frontend:
+
+:param GNU: List of flags for GNU compiler frontend (Gcc, G++)
+:param MSVC: List of flags for MSVC compiler frontend (MSVC, ClangCl)
+
+Detected compiler frontend flags are then outputed in the ``OUTPUT``
+parameter.
+
+Optional ``FILTER`` parameter filter outputed flags with check_cxx_compiler_flag.
+
+.. warning:: When ``FILTER`` option is activated, definition should by passed with
+-D prefix to be valid compiler parameter.
+
+Example
+^^^^^^^
+
+.. code-block:: cmake
+  CXX_FLAGS_BY_COMPILER_FRONTEND(
+    GNU -Wno-conversion -Wno-comment -Wno-self-assign-overloaded
+    MSVC  "/bigobj"
+    OUTPUT COMPLIE_OPTIONS
+    FILTER)
+#]=======================================================================]
+
+function(CXX_FLAGS_BY_COMPILER_FRONTEND)
+  set(options FILTER)
+  set(oneValueArgs OUTPUT)
+  set(multiValueArgs GNU MSVC)
+  cmake_parse_arguments(ARGS "${options}" "${oneValueArgs}" "${multiValueArgs}"
+                        ${ARGN})
+
+  # Before CMake 3.14, the CMAKE_CXX_COMPILER_FRONTEND_VARIANT doesn't exists.
+  # Before CMake 3.26, the CMAKE_CXX_COMPILER_FRONTEND_VARIANT doesn't exists
```

Comment:
There is something strange with those 2 comments

Reply:
The comment line 115 continue line 116. The cmake-format has split the line.

I can rewrite it to make it clearer.